File: frontend/admin-portal/src/types/job.ts
export interface Job {
  id: number;
  title: string;
  company: string;
  location?: string;
  country?: string;
  city?: string;
  workType?: string;
  applyUrl: string;
  source?: string;
  jobId?: string;
  postedDate?: string;
  createdAt: string;
  updatedAt: string;
  isActive: boolean;
  isManualEntry: boolean;
  isVisibleToUsers: boolean;
  description?: string;
  salaryRange?: string;
  tags?: string;
}

export interface CreateJobDto {
  title: string;
  company: string;
  location?: string;
  country?: string;
  city?: string;
  workType?: string;
  applyUrl: string;
  source?: string;
  postedDate?: string;
  description?: string;
  salaryRange?: string;
  tags?: string;
  isVisibleToUsers?: boolean;
}

export interface UpdateJobDto extends Partial<CreateJobDto> {
  isActive?: boolean;
  isVisibleToUsers?: boolean;
}

export interface PaginatedResponse<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface DashboardStats {
  totalJobs: number;
  activeJobs: number;
  visibleJobs: number;
  hiddenJobs: number;
  manualEntries: number;
  jobsAddedToday: number;
  jobsByCountry: Record<string, number>;
  jobsByWorkType: Record<string, number>;
  jobsBySource: Record<string, number>;
}

// Filter types
export interface BlockedCompany {
  id: number;
  companyName: string;
  reason?: string;
  createdAt: string;
  isActive: boolean;
}

export interface BlockedKeyword {
  id: number;
  keyword: string;
  reason?: string;
  createdAt: string;
  isActive: boolean;
}

export interface CreateBlockedCompanyDto {
  companyName: string;
  reason?: string;
}

export interface CreateBlockedKeywordDto {
  keyword: string;
  reason?: string;
}

// Fetch types
export interface FetchOptions {
  fetchGreenhouse?: boolean;
  fetchLever?: boolean;
  fetchWorkable?: boolean;
  fetchJooble?: boolean;
  fetchRemoteOk?: boolean;
  fetchRemotive?: boolean;
  fetchHimalayas?: boolean;
  fetchJobicy?: boolean;
  joobleMaxPages?: number;
}

export interface FetchAndSyncResult {
  message: string;
  fetchResult: {
    totalFetched: number;
    afterDeduplication: number;
    savedToMainApi: number;
    durationSeconds: number;
    sourceStats?: Record<string, number>;
  };
  syncResult: {
    imported: number;
    skipped: number;
    total: number;
  };
}

File: dotnet/EgyptTechJobsAdmin/Models/Entities/Job.cs
using System.ComponentModel.DataAnnotations;

namespace EgyptTechJobsAdmin.Models.Entities;

public class Job
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(500)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string Company { get; set; } = string.Empty;

    [MaxLength(200)]
    public string? Location { get; set; }

    [MaxLength(100)]
    public string? Country { get; set; }

    [MaxLength(100)]
    public string? City { get; set; }

    [MaxLength(50)]
    public string? WorkType { get; set; } // Remote, Hybrid, Onsite

    [Required]
    [MaxLength(1000)]
    public string ApplyUrl { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Source { get; set; }

    [MaxLength(200)]
    public string? JobId { get; set; } // External job ID

    public DateTime? PostedDate { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public bool IsActive { get; set; } = true;

    public bool IsManualEntry { get; set; } = false;

    public bool IsVisibleToUsers { get; set; } = true; // Controls visibility on public site

    [MaxLength(5000)]
    public string? Description { get; set; }

    [MaxLength(500)]
    public string? SalaryRange { get; set; }

    [MaxLength(500)]
    public string? Tags { get; set; } // Comma-separated tags
}
